### DISCUSSION TITLE
calcGrains detection issue

### DISCUSSION THREAD

**Discussion by Fjvs1** [mtex-toolbox/mtex#1226] [Ask Anything] [answered]
Hi !
I'm in trouble with calcGrains 'threshold' or 'angle' parameter. I need to define grains delimitation with a 10°angle.
I tried `grains = calcGrains(ebsd('indexed'),'boundary','tight','threshold',[2*degree,10*degree])` and also `grains = calcGrains(ebsd('indexed'),'angle',10*degree)`, but the result is very different from others softwares. 

To obtain the same detection than those from OIM, Esprit-Bruker or Atex with a 10° threshold, I have to set MTEX's one on 7° or less .It is disabling for the next analysis whom need grains detection. 

Is it normal ?  Thanks for your reply !

**[MTEX with 10° threshold]**
![MTEX 10](https://user-images.githubusercontent.com/87763359/134665302-74ff4e2f-1b92-43d4-885c-7b79a5dfa3e0.png)
**[Esprit with 10° threshold]**
![Bruker](https://user-images.githubusercontent.com/87763359/134665457-58719e2b-8b1d-426a-be7c-1a84b3347820.png)

**Comment by nyyssont** [comment]
It does look like the Esprit grains are a bit more finely segmented. Do you have any details on how the segmentation is conducted in that software? I am unfamiliar with Esprit so it is a bit difficult to say what's wrong specifically.

I can say that a large amount even high-angle subgrain boundaries can sometimes be present in MTEX grains. The calcGrains command is simple in the sense that if two areas can be connected by even a thin "bridge" of low-angle misorientations, calcGrains will then make a single grain out of these areas.

If you are interested in the subgrain boundaries, they should be resolvable when you use the threshold command. Have you tried the instructions on this page:
https://mtex-toolbox.github.io/SubGrainBoundaries.html

**Reply by Fjvs1** [reply]
The Esprit's représentation displays inner-boundary, that's why it's looks like more segmented. I can add bandcontract with transparency on MTEX, it will be nearly the same segmentation. 

My problem is that ATEX, OIM and Esprit (3 differents softwares) are in agreement regarding to grains detection with a threshold of 10°.

When I try 10° on MTEX, many boundaries are not detected that lead to bigger grains. I need to put 7° to obtain something near other results.
For instance, here is OIM detection ( same as Esprit for 10°) and grains detetction by MTEX for 10°, with a grain color depending on the size. Some big grains (in red) are artefacts. 

![OIM](https://user-images.githubusercontent.com/87763359/134695842-7f194ed9-4601-4cae-95ee-999a4432d07d.png)
![MTEX](https://user-images.githubusercontent.com/87763359/134698106-84192cde-d522-45b8-977e-d274e3a9550a.png)

**Comment by nyyssont** [comment, marked as answer]
One special feature in calcGrains - I don't know if something similar has been implemented in other software - is the ability to find neighboring orientation pixels across unindexed regions. This is done by means of Voronoi decomposition.

If desired, Voronoi decomposition can be omitted by supplying the option `'unitCell'`. Could you try

```
grains = calcGrains(ebsd('indexed'),'angle',10*degree,'unitCell')
```

and see if you get a different result?

**Reply by kilir** [reply]
Hi,
you only consider `'indexed'` instead of the entire dataset. Does OIM do the same? Otherwise also use the whole dataset in mtex fox a comparision.
Hope that helps.
cheers, Rüdiger

**Reply by Fjvs1** [reply]
> Hi,
> you only consider `'indexed'` instead of the entire dataset. Does OIM do the same? Otherwise also use the whole dataset in mtex fox a comparision.
> Hope that helps.
> cheers, Rüdiger

Hi ! That is the point !
Everything is fine now without `indexed` Thanks a lot.
